puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/tarballsRelease.groovy

```diff
@@ -0,0 +1,106 @@
+pipeline {
+    agent { label 'admin && sshkey' }
+
+    options {
+        timestamps()
+        timeout(time: 2, unit: 'HOURS')
+        disableConcurrentBuilds()
+        ansiColor('xterm')
+    }
+
+    def version = env.getProperty('version')
+    def major_version = env.getProperty('major_version')
+    def ruby_ver = '2.4.0'
+
+    stages {
+        stage('Parallel') {
+
+            parallel {
+                stage('Foreman') {
+                    steps {
+                        verify_tag('foreman', version)
+                        build_tarball('foreman', version, ruby_ver)
+                    }
+                }
+                stage('Smart Proxy') {
+                    steps {
+                        verify_tag('smart-proxy', version)
+                        build_tarball('smart-proxy', version, ruby_ver)
+                    }
+                }
+                stage('Foreman SELinux') {
+                    steps {
+                        verify_tag('foreman-selinux', version)
+                        build_tarball('foreman-selinux', version, ruby_ver)
+                    }
+                }
+                stage('Foreman Installer') {
+                    steps {
+                        verify_tag('foreman-installer', version)
+                        build_tarball('foreman-installer', version, ruby_ver)
+                    }
+                }
+            }
+        }
+    }
+
+    post {
+        success {
+            build job: 'release_packages', parameters: [
+                [$class: 'StringParameterValue', name: 'version', value: version],
+                [$class: 'StringParameterValue', name: 'major_version', value: major_version]
+            ]
+        }
+    }
+}
+
+void verify_tag(project, version) {
+    deleteDir()
+    dir(project) {
+        git url: "https://github.com/theforeman/${project}", branch: 'develop'
+        sh """
+        if [ \"$(git tag -l ${version} | wc -l)\" -ne 1 ]; then
```

Comment:
I think we can reduce this to simply this line of code with a `sh`. If it returns 1, the `sh` will automatically error out the step which is the behavior I think we want.

Reply:
@komidore64 This comment got lost in the updated push.

Reply:
updated